uniffi/src/bindings/kotlin/templates/Helpers.kt

```diff
@@ -0,0 +1,28 @@
+// A handful of classes and functions to support the generated data structures.
+// This would be a good candidate for isolating in its own ffi-support lib.
+
+abstract class FFIObject {
+    internal val isDestroyed = AtomicBoolean(false)
+
+    open fun destroy() {
+        isDestroyed.set(true)
```

Comment:
I'm wondering if we should throw an error here if the object has already been destroyed. Can't think of a concrete reason why, but one thought I'm having is if someone calls the destroy multiple times, I'm guessing we would give them an error, but is it better to error here? And what kind of error would they get if the call goes through the FFI?

I guess we could do what the other bindings do, would Swift/Python complain if you call the destructor twice? Or would they just pass it into the FFI regardless?

Reply:
As @rfk says, the `ConcurrentHandleMap` should handle that for us. 

I'm torn here: should we be adding `isDestroyed` so we can encourage/enable defensive programming? Or should we make it idempotent? Or should we just go with what we've got, and review ergonomics once we've started using it?

Reply:
I like how you've done it in the PR, making `destroy` idempotent, and throwing if you try to call any methods after destroying the class. It feels like that'll be easier for a-c or whoever to wire it up to `Cleaner` if they want, without having to do additional state tracking.